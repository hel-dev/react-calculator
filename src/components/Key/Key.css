.key-container {
    align-items: center;
    display: flex;
    height: 25%;
    justify-content: center;
    transition: background-color 0.3s linear;
  }
  
  .key-container:hover {
    cursor: pointer;
  }
  
  [type="operator-key"] {
    background-color: var(--action-key-color);
    width: 100%;
  }
  
  [type="operator-key"]:hover {
    background-color: var(--action-key-color-hover);
  }
  
  [type="number-key"] {
    background-color: var(--number-key-color);
    width: calc(100%/3);
  }
  
  [type="number-key"]:hover {
    background-color: var(--number-key-color-hover);
  }
  
  [type="submit-key"] {
    background-color: var(--submit-key-color);
    height: 100%;
    width: 100%;
  }
  
  [type="submit-key"]:hover {
    background-color: var(--submit-key-color-hover);
  }
  
  .key-value {
    color: var(--key-text-color);
    font-family: var(--main-font);
    font-size: var(--key-text-size);
    font-weight: var(--key-text-weight);
  }