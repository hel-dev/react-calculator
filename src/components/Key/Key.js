import React from 'react';
import PropTypes from 'prop-types';
import calculator from 'services/calculator'
import services from 'services'

import './Key.css';

const Key = ({ onClick, type, value }) => 
    <div 
        className={`key-container ${type}`}
        onClick={onClick}
    >
        <p className="key-value">
            {value}
        </p>
    </div>;

Key.propTypes = {
    type: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired
}

export default Key;