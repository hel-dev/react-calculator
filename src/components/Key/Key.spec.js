import React from 'react';
import { shallow } from 'enzyme';
import Key from './Key';

describe('Key', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Key
        onClick={jest.fn()}
        type={''}
        value={''}
      />
    );
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the value of Key value', () => {
    wrapper.setProps({ value: 'test' });
    expect(wrapper.text()).toEqual('test');
  });
});