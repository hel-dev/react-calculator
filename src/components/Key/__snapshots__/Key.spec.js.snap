// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key should render correctly 1`] = `
<div
  className="key-container "
  onClick={[MockFunction]}
>
  <p
    className="key-value"
  />
</div>
`;
